### PR TITLE
DIRECTOR: Preload Stxts in Score

### PR DESCRIPTION
**Pull request by tobiatesan**
I've opened #943, which I branched off this. In case you like also how the preloading in textCast is done, please just merge that one. Thanks.
-----------------
This preloads Stxts in Score and leaves them there.
If you like where this is going so far, you'll soon get also cached MacTexts.

### CODE REVIEW

File: engines/director/frame.h [hunk 1]
```diff
@@ -24,6 +24,7 @@
 #define DIRECTOR_FRAME_H
 
 #include "graphics/managed_surface.h"
+#include "stxt.h"
```
Comment (sev-) [inline engines/director/frame.h]: incorrect. must be a full path

File: engines/director/score.h [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #include "director/archive.h"
 #include "director/cast.h"
 #include "director/images.h"
+#include "stxt.h"
```
Comment (sev-) [inline engines/director/score.h]: Same here

File: engines/director/stxt.h [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#ifndef DIRECTOR_STXT_H
+#define DIRECTOR_STXT_H
+
+#include "director.h"
```
Comment (sev-) [inline engines/director/stxt.h]: Same here

File: engines/director/stxt.h [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+namespace Director {
+
+class Stxt {
+public:
+	Stxt (Common::SeekableSubReadStreamEndian &textStream) {
```
Comment (sev-) [inline engines/director/stxt.h]: Please move to a .cpp file

File: engines/director/score.h [hunk 5]
```diff
@@ -133,6 +134,7 @@ class Score {
 	Common::HashMap<int, BitmapCast *> *_loadedBitmaps;
 	Common::HashMap<int, ShapeCast *> *_loadedShapes;
 	Common::HashMap<int, ScriptCast *> *_loadedScripts;
+	Common::HashMap<int, const Stxt*> *_loadedStxts;
```
Comment (sev-) [inline engines/director/score.h]: Incorrect formatting

File: engines/director/score.cpp [hunk 6]
```diff
@@ -92,6 +92,7 @@ Score::Score(DirectorEngine *vm) {
 	_loadedButtons = new Common::HashMap<int, ButtonCast *>();
 	_loadedShapes = new Common::HashMap<int, ShapeCast *>();
 	_loadedScripts = new Common::HashMap<int, ScriptCast *>();
+	_loadedStxts = new Common::HashMap<int, const Stxt*>();
```
Comment (sev-) [inline engines/director/score.cpp]: Incorrect formatting

**Comment by tobiatesan** [comment]
Sorry, I shouldn't have let those slip by.

**Comment by tobiatesan** [comment]
Fixed.

**Comment by tobiatesan** [comment]
@sev- I also took the liberty of adding 5a28d22.

**Comment by tobiatesan** [comment]
**I've opened #941, which I branched off this. In case you like also how the preloading in textCast is done, please just merge that one. Thanks.**

**Comment by sev-** [comment]
Closing in favor of #944

**Pull request closed without merging**